### PR TITLE
Install and build on macOS

### PR DESCRIPTION
**Pull request by n-b**
I don’t know if it _runs_ correctly, but at least it builds on my machine.

The most annoying thing is for openssl, which required me to add environment variables. I would have preferred adding keys in Cargo.toml, as per http://stackoverflow.com/a/38946292. I only got cargo to complain about `warning: unused manifest key`, not 100% sure why.

### CODE REVIEW

File: tools/install.sh [hunk 1]
```diff
@@ -1,18 +1,26 @@
...
-if [ -z $DATABASE_URL ]; then
-  echo "DATABASE_URL is not set. Aborting."
-  exit 1;
+# libsqlite3-dev
+if [[ "$OSTYPE" == "linux-gnu" ]]; then
+	sudo apt update
+	sudo apt install libsqlite3-dev
+# elif [[ "$OSTYPE" == "darwin"* ]]; then # macOS
+	# libsqlite3-dev is installed by default on any recent macOS
 fi
```
Comment (rlustin) [inline tools/install.sh]: Could you add something like:
```bash
 else
    echo "error can't install package `libsqlite3-dev`, unknown OS $OSTYPE"
    exit 1;
 fi
```

File: tools/install.sh [hunk 2]
```diff
@@ -1,18 +1,26 @@
 #!/bin/bash -ex
 
-if [ -z $DATABASE_URL ]; then
-  echo "DATABASE_URL is not set. Aborting."
-  exit 1;
+# libsqlite3-dev
+if [[ "$OSTYPE" == "linux-gnu" ]]; then
+	sudo apt update
+	sudo apt install libsqlite3-dev
```
Comment (rlustin) [inline tools/install.sh]: We’ll need to choose the good packaging system, but that’s the subject of another PR.

Maybe something like http://stackoverflow.com/a/19520888

File: tools/install.sh (hunk 1 above)
Comment (n-b) [inline tools/install.sh]: done

**Comment by rlustin** [comment]
Thank you for this PR!

It’ll help a lot new contributors :tada:

**Pull request merged**